build_tools/benchmarks/common/benchmark_config_test.py

```diff
@@ -43,26 +46,27 @@ def test_build_from_args(self):
 
     config = BenchmarkConfig.build_from_args(args=args, git_commit_hash="abcd")
 
-    per_commit_tmp_dir = os.path.join(self.tmp_dir.name, "abcd")
+    per_commit_tmp_dir = os.path.realpath(
+        os.path.join(self.tmp_dir.name, "abcd"))
     expected_trace_capture_config = TraceCaptureConfig(
         traced_benchmark_tool_dir=self.traced_tool_dir,
-        trace_capture_tool=self.trace_capture_tool.name,
+        trace_capture_tool=os.path.realpath(self.trace_capture_tool.name),
         capture_tarball=os.path.realpath("capture.tar"),
         capture_tmp_dir=os.path.join(per_commit_tmp_dir, "captures"))
-    self.assertEqual(
-        config,
-        BenchmarkConfig(root_benchmark_dir=os.path.join(self.build_dir.name,
-                                                        "benchmark_suites"),
-                        benchmark_results_dir=os.path.join(
-                            per_commit_tmp_dir, "benchmark-results"),
```

Comment:
This test got messy to update, since `BenchmarkConfig` calls `os.path.realpath` in many places.

This led to errors when comparing paths like
```
'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\tmpm6qhk0oy\\benchmark_suites'
'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpm6qhk0oy\\benchmark_suites'
```

Could we just add a way to "assert equal, after canonicalizing all paths"?

Reply:
If we stored `pathlib.Path` objects instead of strings, the comparisons should be safer too.

Reply:
I'm fine with the current solution. We should refactor the tool to take pathlib.Path instead eventually. Can you file a bug?

Reply:
https://github.com/iree-org/iree/issues/11055